src/ghci/loaded_module.rs

```diff
@@ -0,0 +1,96 @@
+use std::borrow::Borrow;
+use std::fmt::Display;
+use std::hash::Hash;
+use std::hash::Hasher;
+
+use camino::Utf8Path;
+
+use crate::normal_path::NormalPath;
+
+/// Information about a module loaded into a `ghci` session.
+///
+/// Hashing and equality are determined by the module's path alone.
+#[derive(Debug, Clone, Eq)]
+pub struct LoadedModule {
+    /// The module's source file, like `src/My/Cool/Module.hs`.
+    path: NormalPath,
+
+    /// The module's dotted name, like `My.Cool.Module`.
+    ///
+    /// This is present if and only if the module is loaded by name.
+    ///
+    /// Entries in `:show targets` can be one of two types: module paths or module names (with `.` in
+    /// place of path separators). Due to a `ghci` bug, the module can only be referred to as whichever
+    /// form it was originally added as (see below), so we use this to track how we refer to modules.
+    ///
+    /// See: <https://gitlab.haskell.org/ghc/ghc/-/issues/13254#note_525037>
+    name: Option<String>,
+}
+
+impl LoadedModule {
+    /// Create a new module, loaded by path.
+    pub fn new(path: NormalPath) -> Self {
+        Self { path, name: None }
+    }
+
+    /// Create a new module, loaded by name.
+    pub fn with_name(path: NormalPath, name: String) -> Self {
+        Self {
+            path,
+            name: Some(name),
+        }
+    }
+
+    /// Get the module's source path.
+    pub fn path(&self) -> &NormalPath {
+        &self.path
```

Comment:
Similar question here: what's the advantage of doing `loadedModule.path()` instead of `loadedModule.path`?